server/src/test/java/org/apache/calcite/avatica/server/CustomAuthHttpServerTest.java

```diff
@@ -175,63 +126,165 @@ public void testCustomBasicImpersonationConfigWithDisallowedUser() throws Except
     }
   }
 
+  @Test
+  public void testCustomConfigDisallowsWithHandlerMethod() {
+    AvaticaServerConfiguration configuration = new CustomBasicImpersonationConfig();
+    server = new HttpServer.Builder()
+            .withCustomAuthentication(configuration)
+            .withHandler(Mockito.mock(AvaticaHandler.class))
+            .withPort(0)
+            .build();
+    try {
+      server.start();
+      fail("Expected an exception");
+    } catch (IllegalStateException e) {
+      String assertString = "Handlers and SSLFactory cannot be configured with "
+              + "HTTPServer Builder when using CUSTOM Authentication Type";
+      assertThat(e.getMessage(), containsString(assertString));
```

Comment:
I'm not a fan of tests that assert a certain error message. IMO, if it's important enough to validate that we hit this exact code path (and not another similar), we should have a named exception.

I think just validating that the `ISE` was thrown is sufficient.